sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp

```diff
@@ -374,7 +374,14 @@ namespace Azure { namespace Storage { namespace Blobs {
     /**
      * @brief Optional conditions that must be met to perform this operation.
      */
-    BlobContainerAccessConditions AccessConditions;
+    struct : public LeaseAccessConditions
```

Comment:
Not single purpose, and please add change log.

Reply:
It is single purpose. The main purpose of this PR is to correct storage exception types. In the past, we throw exception on unsupported access condition. Now that we removed the unsupported access condition, we don't need to throw any more

Reply:
Plus I don't think this need to be reflected in changelog.

Reply:
This option used to have `IfModifiedSince` and `IfUnmodifiedSince`, but now only `IfModifiedSince` right? Why is that not something needs to be reflected in changelog?

Reply:
`IfUnmodifiedSince` never works. It throws in the past, now it crashes. Why would anybody care about it?

Reply:
No, it does not crash now, it is a compiler error now. Are you saying that the precondition check changed to compiler error is not breaking change?

Reply:
I may be able to accept adding it into changelog? But is it a breaking change? I don't think so.

Reply:
I believe it is either a breaking change, or we shouldn't mention it at all.
Consider if the customer assigns a `BlobContainerAccessConditions` everywhere in the options, it used to work if `IfUnmodifiedSince` is null, but not now for this API.

Reply:
well, okay